### PR TITLE
feat(healthcheck): adds api route health check functionality

### PR DESCRIPTION
**Pull request by eve-horne**
# Changes introduced
Adds a health-check endpoint listing status of all routes at '/health'.
From now on, routes need to use the 'setRouteHealth()' method to set their status when being used.

## Related issue(s):

- CSE-_279

## Contributor checklist

- [x] Wrote/updated tests for introduced changes
- [ ] ~~Added new stories for created/modified components (if applicable)~~
- [x] Updated Postman library for created/modified routes (if applicable)
- [x] Verified that there are no issues in CircleCI
- [x] Assigned yourself to the PR
- [x] Removed `WIP:` from the PR title
- [x] Requested review from the SB Reviewers team

## Reviewer checklist

- [x] Assigned yourself to the PR
- [ ] Read through the changes in the `Files changed` tab
- [ ] Verified that tests were written/modified
- [ ] ~~Verified that stories were written/modified (if applicable)~~
- [ ] Verified that Postman was updated (if applicable)
- [ ] Checked out the branch and tested changes locally

### CODE REVIEW

File: src/routes/health/health.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      send: jest.fn(),
+      status: jest.fn()
+    };
+    req = {};
+  });
+
+  it('sets route health', () => {
+    status = Health.busy;
+    req.path = '/init';
+    setRouteHealth(status, <Request>req);
```
Comment (danvanhorn) [inline src/routes/health/health.spec.ts:31]: We're not actually checking that anything happens here. We'll need to actually confirm that this function performed the correct task.

File: src/routes/health/health.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  it('returns a response', () => {
+    healthCheck(<Request>req, <Response>res);
+    expect(res.send).toBeCalledTimes(1);
+    expect(res.status).toHaveBeenCalledWith(200);
```
Comment (danvanhorn) [inline src/routes/health/health.spec.ts:47]: We should `expect.assertions(2)` here.

File: src/routes/target/target.spec.ts [hunk 3]
```diff
@@ -1,6 +1,17 @@
 import { handler as target } from './index';
 import { Request, Response } from 'express';
 
+jest.mock('../health', () => ({
```
Comment (danvanhorn) [inline src/routes/target/target.spec.ts]: This mock is repeated in eahc of the routes. If you put a `__mocks__` directory adjacent to `src/routes/health/index.ts` and put that mock inside, `jest` will automatically imported that mocked module during our tests.

File: src/server/index.ts [hunk 4]
```diff
@@ -70,6 +71,11 @@ export class Server implements IServer {
         dbName: 'cse'
       })
     };
+    router.stack.forEach(endpoint => {
+      // The express router stack is of type any[]
+      // tslint:disable-next-line
```
Comment (danvanhorn) [inline src/server/index.ts]: `//tslint:disable-next-line: no-any no-unsafe-any`
This should illustrate the point that the endpoint is of type `any`. We can't really do much about certain any types being returned from libraries, so the occasional line disabling is necessaary.

File: src/routes/health/index.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Request } from 'express';
```
Comment (danvanhorn) [inline src/routes/health/index.ts:1]: I think this is good. Eventually we will want to create health endpoints for each of our containers that we can loop through, check and set the corresponding endpoint availability. As of right now, the endpoint health is set at the beginning and the end of the request, which works great for the long running `init/` endpoint. For `search/` and `filter/`, we should check if the elasticsearch and mongodb containers are running and that will more accurately represent the health of our endpoints.

File: src/routes/health/health.spec.ts (hunk 1 above)
Comment (eve-horne) [reply src/routes/health/health.spec.ts:31]: This has been resolved in the latest changes.

File: src/routes/health/health.spec.ts (hunk 2 above)
Comment (eve-horne) [reply src/routes/health/health.spec.ts:47]: This has been resolved in the latest changes

File: src/routes/target/target.spec.ts (hunk 3 above)
Comment (eve-horne) [reply src/routes/target/target.spec.ts]: This has been resolved in the latest changes.

File: src/server/index.ts (hunk 4 above)
Comment (eve-horne) [reply src/server/index.ts]: This has been resolved in the latest changes.

**Review by danvanhorn — APPROVED** [review]
This looks good. Good job increasing test coverage again as well.

**Pull request merged**